app/javascript/redux/greetings/greetings.js

```diff
@@ -0,0 +1,20 @@
+const initState = '';
+
+const greetingReducer = (state = initState, action) => {
+  switch (action.type) {
+    case 'GET_GREETING':
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+export const fetchgreeting = () => async (dispatch) => {
+  await fetch('http://127.0.0.1:3000/api/greetings')
+    .then((response) => response.json())
+    .then((data) => {
+      dispatch({ type: 'GET_GREETING', payload: data });
+    })
+};
```

Comment:
- [ ] When I start the application, I get an error that it can't fetch from this URL due to the CORS origin issue. There are two ways you can solve this issue:
- You can change the URL to `/api/greetings` and rails will automatically fetch the data from the endpoint.
- The second option is to include `gem 'rack-cors'` in your Gemfile, run `bundle install` and in the  **_config/initializers/cors.rb file_**, do the following: 

```
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins '*'

    resource '*',
      headers: :any,
      methods: [:get, :post, :put, :patch, :delete, :options, :head]
  end
end
```

#### N.B It is alright to do it this way in a development setting, but if you were to ship this code to production, you would have your Rack::Cors look this way for security reasons:

```
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'example.com'

    resource '*',
      headers: :any,
      methods: [:get, :post, :put, :patch, :delete, :options, :head]
  end
end
```

Reply:
## Hi @Gambit142 
Thank you for clarifying the issue.
I will go with the first way.

Reply:
Hi @Bondok6, great one, nice execution. 👍